### ISSUE TITLE
target_normalizer doesn't work

### ISSUE THREAD

**Issue by LumingSun** [jdb78/pytorch-forecasting#410]
- PyTorch-Forecasting version: 0.8.4
- PyTorch version: 1.8.0
- Python version: 3.8.8
- Operating System: Debian 9

### Expected behavior

I followed the tutorial in [Demand forecasting with the Temporal Fusion Transformer](https://pytorch-forecasting.readthedocs.io/en/latest/tutorials/stallion.html), and after I created the TimeSeriesDataSet (with target_normalizer as the example code) and dataloader, I run
```
for x,y in iter(train_dataloader):
    print(y)
```
The y (time series target) should have been normalized with GroupNormalizer in "standard" method.

### Actual behavior
However, the printed y was the same as it was in raw dataset. I tried with other normalizers and methods but still no change.

### Code to reproduce the problem
Just run
```
for x,y in iter(train_dataloader):
    print(y)
```
after the 5th cell in the tutorial.

**Comment by jdb78** [comment]
The `y` should not change as we score against it. The normaliser changes the input, i.e. the part of the `x["encoder_cont"]` that represents the time series and how the prediction is de-normalised.

**Comment by LumingSun** [comment]
Thank you for your reply. I understand that `y` is not scaled or normalized now. But I still have some problem with normalizer or scaler. 
In my scenario, I have data with only three attributes: groupID, date, and number. But the target number I need to predict is quite large (larger than 1e7). I want to normalize the target so I added `GroupNormalizer` to `target_normalizer`, but I got huge training loss (larger than 1e10), so I suspect that the target time series is not rescaled or normalized. 
I printed the `x` of dataloader, and I found both the `encoder_target` and `decoder_target` are the same as raw data, and only `encoder_cont` and `decoder_cont` are normalized. Does this mean the `target_normalizer` only normalizer features but not labels (i.e. the target time series)?

**Comment by jdb78** [comment]
Yes. Features and predictions are normalized and de-normalized. The labels must not be normalized to correctly calculate metrics. You should see if things are working by checking individual plots logged on the tensorboard. You can check the normalization factors in `x["target_scale"]`. It has two dimensions for `mean` and `scale`.

**Comment by LumingSun** [comment]
> Yes. Features and predictions are normalized and de-normalized. The labels must not be normalized to correctly calculate metrics. You should see if things are working by checking individual plots logged on the tensorboard. You can check the normalization factors in `x["target_scale"]`. It has two dimensions for `mean` and `scale`.

The predictions should be de-normalized when calculating metrics with the labels. But during training, they are supposed to be normalized (usually to numbers less than 10), and if so, the loss shouldn't be so large. I checked the `x["target_scale"]`, it seems no problem. But my loss during training is extremely large and I just suspect the normalizer didn't work. I'll do some digging, thank you!

**Comment by breearie** [comment]
> > Yes. Features and predictions are normalized and de-normalized. The labels must not be normalized to correctly calculate metrics. You should see if things are working by checking individual plots logged on the tensorboard. You can check the normalization factors in `x["target_scale"]`. It has two dimensions for `mean` and `scale`.
> 
> The predictions should be de-normalized when calculating metrics with the labels. But during training, they are supposed to be normalized (usually to numbers less than 10), and if so, the loss shouldn't be so large. I checked the `x["target_scale"]`, it seems no problem. But my loss during training is extremely large and I just suspect the normalizer didn't work. I'll do some digging, thank you!

How did you finally solve the problem of excessive loss?  Thanks in advance!
